src/Microsoft.Framework.ConfigurationModel/Configuration.cs

```diff
@@ -138,12 +138,15 @@ public IConfiguration GetSubKey(string key)
 
         public IConfigurationSourceRoot Add(IConfigurationSource configurationSource)
         {
-            configurationSource.Load();
-            return AddLoadedSource(configurationSource);
+            return Add(configurationSource, load: true);
         }
 
-        internal IConfigurationSourceRoot AddLoadedSource(IConfigurationSource configurationSource)
+        public IConfigurationSourceRoot Add(IConfigurationSource configurationSource, bool load)
```

Comment:
Can we add a doc comment to this, perhaps warning that it's for testing scenarios only?